modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/ExportImportObjectFolderTest.java

```diff
@@ -0,0 +1,310 @@
+
+	@Before
+	public void setUp() throws Exception {
+		_user = TestPropsValues.getUser();
+
+		ObjectFolderResource.Builder builder =
+			_objectFolderResourceFactory.create();
+
+		_objectFolderResource = builder.user(
+			_user
+		).build();
+	}
+
+	@Test
+	public void testImportObjectFoldersWithObjectDefinitions()
+		throws Exception {
+
+		// Import and export an empty object folder
+
+		_testImportExportObjectFolder(
+			"EMPTY_OBJECT_FOLDER", "empty_object_folder.json",
+			"EmptyObjectFolder", "empty_object_folder.json", 0, false);
+
+		// Import an object folder that has a linked object that doesn't exist
+
+		_testImportExportObjectFolder(
+			"OBJECT_FOLDER", "object_folder.json", "ObjectFolder",
+			"object_folder.json", 2, true);
+
+		// Import an object folder with invalid folder items
+
+		_testImportExportObjectFolder(
+			"EMPTY_OBJECT_FOLDER", "object_folder_invalid_items.json",
+			"EmptyObjectFolder", "empty_object_folder.json", 0, true);
+
+		// Import an object folder that moves an ObjectDefinition during import
+
+		_testImportExportObjectFolder(
+			"LINKED_OBJECT_FOLDER", "linked_object_folder.json",
+			"LinkedObjectFolder", "linked_object_folder.json", 2, false);
+
+		// Import an object folder with conflicting ERC, but with less
+		// ObjectDefinitions
+
+		_testImportExportObjectFolder(
+			"LINKED_OBJECT_FOLDER", "empty_object_folder.json",
+			"EmptyObjectFolder", "empty_object_folder.json", 0, true);
+	}
+
+	private ObjectFolder _getObjectFolder(String name) throws Exception {
+		Page<ObjectFolder> page = _objectFolderResource.getObjectFoldersPage(
+			name, Pagination.of(1, 1));
+
+		List<ObjectFolder> items = (List<ObjectFolder>)page.getItems();
+
+		return items.get(0);
+	}
+
+	private ThemeDisplay _getThemeDisplay() throws Exception {
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		Layout layout = new LayoutImpl();
+
+		layout.setType(LayoutConstants.TYPE_CONTROL_PANEL);
+
+		themeDisplay.setLayout(layout);
+
+		themeDisplay.setScopeGroupId(TestPropsValues.getGroupId());
+		themeDisplay.setSiteDefaultLocale(LocaleUtil.US);
+		themeDisplay.setUser(_user);
+
+		return themeDisplay;
+	}
+
+	private List<String> _getUncategorizedObjectFolderItemERCs()
+		throws Exception {
+
+		ObjectFolder uncategorizedObjectFolder = _getObjectFolder(
+			"Uncategorized");
+
+		return TransformUtil.transform(
+			ListUtil.fromArray(
+				uncategorizedObjectFolder.getObjectFolderItems()),
+			ObjectFolderItem::getObjectDefinitionExternalReferenceCode);
+	}
+
+	private void _testImportExportObjectFolder(
+			String externalReferenceCode, String sourceFile, String folderName,
+			String compareFile, long assertLength, boolean assertPresence)
+		throws Exception {
+
+		MockMultipartHttpServletRequest mockMultipartHttpServletRequest =
+			new MockMultipartHttpServletRequest();
+
+		Class<?> clazz = getClass();
+
+		byte[] bytes = _file.getBytes(
+			clazz.getResourceAsStream("dependencies/" + sourceFile));
+
```

Comment:
This method is very similar to the `ExportImportObjectDefinitionTest.java`, can we create a `*TestUtil` class and put all this code there? This way we can avoid repeating code.

Feel free to ask if you have any questions